packages/dashboard-frontend/src/store/GitOauthConfig/index.ts

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2018-2021 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { Action, Reducer } from 'redux';
+import common, { api } from '@eclipse-che/common';
+import { AppThunk } from '..';
+import { createObject } from '../helpers';
+import { AUTHORIZED } from '../sanityCheckMiddleware';
+import { container } from '../../inversify.config';
+import { CheWorkspaceClient } from '../../services/workspace-client/cheworkspace/cheWorkspaceClient';
+import { IGitOauth } from './types';
+
+export interface State {
+  isLoading: boolean;
+  gitOauth: IGitOauth[];
+  error: string | undefined;
+}
+
+const cheWorkspaceClient = container.get(CheWorkspaceClient);
+
+export enum Type {
+  REQUEST_GIT_OAUTH_CONFIG = 'REQUEST_GIT_OAUTH_CONFIG',
+  DELETE_OAUTH = 'DELETE_OAUTH',
+  RECEIVE_GIT_OAUTH_CONFIG = 'RECEIVE_GIT_OAUTH_CONFIG',
+  RECEIVE_GIT_OAUTH_CONFIG_ERROR = 'RECEIVE_GIT_OAUTH_CONFIG_ERROR',
+}
+
+export interface RequestGitOauthConfigAction extends Action {
+  type: Type.REQUEST_GIT_OAUTH_CONFIG;
+}
+
+export interface DeleteOauthAction extends Action {
+  type: Type.DELETE_OAUTH;
+  provider: api.GitOauthProvider;
+}
+
+export interface ReceiveGitOauthConfigAction extends Action {
+  type: Type.RECEIVE_GIT_OAUTH_CONFIG;
+  gitOauth: IGitOauth[];
+}
+
+export interface ReceivedGitOauthConfigErrorAction extends Action {
+  type: Type.RECEIVE_GIT_OAUTH_CONFIG_ERROR;
+  error: string;
+}
+
+export type KnownAction =
+  | RequestGitOauthConfigAction
+  | DeleteOauthAction
+  | ReceiveGitOauthConfigAction
+  | ReceivedGitOauthConfigErrorAction;
+
+export type ActionCreators = {
+  requestGitOauthConfig: () => AppThunk<KnownAction, Promise<void>>;
+  revokeOauth: (oauthProvider: api.GitOauthProvider) => AppThunk<KnownAction, Promise<void>>;
+};
+
+export const actionCreators: ActionCreators = {
+  requestGitOauthConfig:
+    (): AppThunk<KnownAction, Promise<void>> =>
+    async (dispatch): Promise<void> => {
+      await dispatch({
+        type: Type.REQUEST_GIT_OAUTH_CONFIG,
+      });
+      const gitOauth: IGitOauth[] = [];
+      try {
+        const oAuthProviders = await cheWorkspaceClient.restApiClient.getOAuthProviders();
+        for (const { name, endpointUrl } of oAuthProviders) {
+          try {
+            await cheWorkspaceClient.restApiClient.getOAuthToken(name);
+            gitOauth.push({
+              name: name as api.GitOauthProvider,
+              endpointUrl,
+            });
+          } catch (e) {
+            // no-op
+          }
+        }
```

Comment:
Requesting tokens for each provider sequentially is not the best approach. Don't you mind refactoring this code to make all requests simultaneously?

Reply:
I have fixed it.